### DISCUSSION TITLE
アバターデコレーションのロールの設定について

### DISCUSSION THREAD

**Discussion by fujisan-sky** [misskey-dev/misskey#14874] [Ideas]
すみません。
2024.10.2-alpha.2　で、個々のアバターデコレーションを使える、ロールの設定が加わったのですが。

これまでのロールの実装と異なりロールが（権限/RolePoliciesの仲介無しに）直接アバターデコレーションにリンクされているようです。
この仕様は従来のロールの運用・操作に対して違和感があると思います。

これまでのロールの運用・操作を踏襲するなら、なんらかの権限/RolePolicies（っぽい何か）を新設して、それを介してアバターデコレーションとロールを連携した方が良いように思います。
もちろん現状の権限/RolePoliciesの実装はスタティックなので、（それをそのまま使うと）自由度が確保できません。
（例えば、SpecialAvatarというフラグを導入していくつかのアバターデコレーションに設定し、RolePolicyにcanUseSpecialAvatarと加えてロール定義時に設定する形だと、簡単に実装できますが１グループしか実現できません。）

つまり、自由度を確保するためには、新たにダイナミックに定義できる権限/RolePolicies（に近い何か？）の概念と、その管理を（たいへんだとは思うのですが）作らないといけないと思います。
要するにPolicyA,PolicyB、…、PolicyX等々を、定義変更削除できる管理機能です。
そして、アバターデコレーションを、その概念と紐づけます（その場合、今のalpha.2の実装のようにアバターデコレーションを複数のロールを紐づけるのでは無く、単一のPolicyX(?)を紐づければ良いと思います）。

そして、複数のロールが同一のPolicyXを利用可能とします。その際のPolicyXの利用可能の可否は、従来のロールの定義と同様の優先度設定が使えると、従来の操作・運用と同様のロールの管理が可能になりましょう。対象が静的なRollPoliciesの一つでは無く、定義されたPolicyXを選択して、それに対するのものになるだけです。（ダケって、、。ごめんなさい、作るのはたいへんですよね。）

また、そうしたPolicies管理を導入できれば、将来的にアバターデコレーション以外の資源（絵文字リアクション、チャンネル等々）の利用権利や、動的な権利（アイデアがありそうです）に対しての、権限の割り当ての際に使えると思います。

余計な事とは思いますが、少しでも参考になれば幸いです。

**Comment by syuilo** [comment]
FYI: 従来からカスタム絵文字の権限設定もこの形式の設定です

**Reply by fujisan-sky** [reply]
ごめんなさい。気が付きませんでした。そういう事だと今更でしたね。
